tests/integration_tests/test_parameter_example.py

```diff
@@ -0,0 +1,298 @@
+            )
+            if grid_format == "egrid":
+                grid.save_EGRID(grid_file)
+            elif grid_format == "grid":
+                grid.save_GRID(grid_file)
+            else:
+                raise ValueError()
+        elif lib == IoLibrary.ERT:
+            egrid = self.data.draw(egrids)
+            self.dims = egrid.shape
+            self.size = self.dims[0] * self.dims[1] * self.dims[2]
+            self.actnum = egrid.global_grid.actnum
+            egrid.to_file(grid_file)
+        else:
+            raise ValueError()
+
+    def _random_values(self, shape):
+        return self.data.draw(
+            arrays(
+                elements=st.floats(min_value=1.0, max_value=10.0, width=32),
+                dtype=np.float32,
+                shape=shape,
+            )
+        )
+
+    def create_field(
+        self,
+        name: str,
+        file_name: str,
+        fformat: FieldFileFormat,
+    ) -> None:
+        lib = self._choose_lib()
+        values = self._random_values(self.dims)
+        self.field_values[file_name] = values
+
+        if lib == IoLibrary.XTGEO:
+            prop = xtgeo.GridProperty(
+                ncol=self.dims[0],
+                nrow=self.dims[1],
+                nlay=self.dims[2],
+                name=name,
+                values=values,
+            )
+            prop.to_file(
+                file_name, fformat="roff" if fformat == "roff_binary" else fformat
+            )
+        elif lib == IoLibrary.RESDATA:
+            if fformat == FieldFileFormat.GRDECL:
+                kw = ResdataKW(name, self.size, ResDataType.RD_FLOAT)
+                data = values.ravel()
+                for i in range(self.size):
+                    kw[i] = data[i]
+                with cwrap.open(file_name, mode="w") as f:
+                    kw.write_grdecl(f)
+            else:
+                # resdata cannot write roff
+                save_field(np.ma.masked_array(values), name, file_name, fformat)
+
+        elif lib == IoLibrary.ERT:
+            save_field(np.ma.masked_array(values), name, file_name, fformat)
+        else:
+            raise ValueError()
+
+    def create_surface(self, file_name: str) -> None:
+        values = self._random_values((2, 5))
+        self.surface_values[file_name] = values
+        xtgeo.RegularSurface(
+            ncol=values.shape[0],
+            nrow=values.shape[1],
+            xinc=1.0,
+            yinc=1.0,
+            values=values,
+        ).to_file(file_name, fformat="irap_ascii")
+
+
+@settings(max_examples=10)
+@given(
+    io_source=st.builds(IoProvider, data=st.data()),
+    grid_format=st.sampled_from(["grid", "egrid"]),
+    summary=summaries(
+        start_date=st.just(datetime(1996, 1, 1)),
+        time_deltas=st.just([1.0, 2.0]),
+        summary_keys=st.just(["FOPR"]),
+    ),
+)
+def test_parameter_example(io_source, grid_format, summary, tmpdir_factory):
+    dir = tmpdir_factory.mktemp("parameter_example")
```

Comment:
Use `tmp_path_factory` as it's preferred and not legacy. https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures

```suggestion
def test_parameter_example(io_source, grid_format, summary, tmp_path_factory, monkeypatch):
    dir = tmp_path_factory.mktemp("parameter_example")
    monkeypatch.chdir(dir)
```